sycl/doc/SharedLibraries.md

```diff
@@ -167,21 +190,99 @@ struct _pi_device_binary_property_struct {
 ```
 
 List of imported symbols is represented as a single property set with name
-`ImportedSymbols` recorded in the `Name` field of property set.
+`SYCL/imported symbols` recorded in the `Name` field of property set.
 Each property in this set holds name of the particular imported symbol recorded
 in the `Name` field of the property.
 List of exported symbols is represented in the same way, except the
-corresponding set has the name `ExportedSymbols`.
+corresponding set has the name `SYCL/exported symbols`.
 
 ### DPC++ runtime changes
 
 DPC++ RT performs *device images collection* task by grouping all device
 images required to execute a kernel based on the list of exports/imports and
 links them together using PI API.
 
-Given that all exports will be arranged to a separate device images without
-kernels it is reasonable to store device images with exports in a separate data
-structure.
+#### Device images collection
+
+DPC++ Runtime class named ProgramManager stores device images using following
+data structure:
+```
+/// Keeps all available device executable images added via \ref addImages.
+/// Organizes the images as a map from a kernel set id to the vector of images
+/// containing kernels from that set.
+/// Access must be guarded by the \ref Sync::getGlobalLock()
+std::unordered_map<SymbolSetId,
+                   std::unique_ptr<std::vector<RTDeviceBinaryImageUPtr>>>
+    m_DeviceImages;
+
+using StrToKSIdMap = std::unordered_map<string_class, SymbolSetId>;
+/// Maps names of kernels from a specific OS module (.exe .dll) to their set
+/// id (the sets are disjoint).
+std::unordered_map<OSModuleHandle, SymbolSetId> m_SymbolSets;
```

Comment:
NIT: `OSModuleHandle`s are no longer needed, as it was clarified by @gmlueck that two device binary images exporting the same function (name) is an error (unless these are two different code forms of the same function). So `OSModuleHandle` could be removed now if it would simplify design/implementation of the dynamic linkning.

Reply:
Yeah, @romanovvlad already mentioned it. However using of `OSModuleHandles`, so some parts (like cache) still depend on this thing. We agreed that once `OSOSModuleHandle` using is removed I'll update the doc accordingly.

Reply:
ok